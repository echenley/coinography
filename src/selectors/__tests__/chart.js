import {
  getChartType,
  getCurrency,
  getGranularity,
  getRawChartData,
  getChartDataLoadingState,
  getCandleChartData,
  getLineChartData,
} from '../chart'

const emptyState = {
  chart: {
    currency: 'BTC-USD',
    data: null,
    granularity: 300,
    loadingState: 'loading',
    requestId: 1,
    type: 'candle',
  },
}

const state = {
  chart: {
    currency: 'BTC-USD',
    data: [
      [1520120700, 11408.47, 11433, 11419, 11433, 6.950522849999999],
      [1520120400, 11402.09, 11432.5, 11431.99, 11418.99, 16.592416949999997],
      [1520120100, 11423.98, 11437, 11423.99, 11432, 8.08159502],
      [1520119800, 11417, 11472.99, 11472.99, 11423.99, 14.742709310000002],
      [1520119500, 11407.15, 11469, 11407.16, 11451.85, 32.78077041000002],
      [1520119200, 11375.93, 11407.16, 11375.94, 11407.16, 32.09318542999999],
      [1520118900, 11365.67, 11375.94, 11365.67, 11375.94, 8.343350529999997],
      [1520118600, 11365.67, 11375.01, 11375.01, 11365.68, 7.895380060000002],
      [1520118300, 11374.99, 11380, 11375, 11375, 9.742431200000002],
      [1520118000, 11374.99, 11375, 11375, 11375, 13.58678584],
    ],
    granularity: 300,
    loadingState: 'success',
    requestId: 1,
    type: 'candle',
  },
}

describe('getChartType', () => {
  it('returns getChartType', () => {
    expect(getChartType(state)).toBe('candle')
  })
})

describe('getCurrency', () => {
  it('returns getCurrency', () => {
    expect(getCurrency(state)).toBe('BTC-USD')
  })
})

describe('getGranularity', () => {
  it('returns getGranularity', () => {
    expect(getGranularity(state)).toBe(300)
  })
})

describe('getRawChartData', () => {
  it('returns getRawChartData', () => {
    expect(getRawChartData(state)).toBe(state.chart.data)
  })
})

describe('getChartDataLoadingState', () => {
  it('returns getChartDataLoadingState', () => {
    expect(getChartDataLoadingState(state)).toBe('success')
  })
})

describe('getCandleChartData', () => {
  it('returns candle chart data', () => {
    expect(getCandleChartData(state, 5)).toMatchSnapshot()
  })

  it('uses default limit of 30 if not provided', () => {
    expect(getCandleChartData(state)).toHaveLength(state.chart.data.length)
  })

  it('returns null if no data', () => {
    expect(getCandleChartData(emptyState, 5)).toBe(null)
  })
})

describe('getLineChartData', () => {
  it('returns line chart data', () => {
    expect(getLineChartData(state, 5)).toMatchSnapshot()
  })

  it('uses default limit of 30 if not provided', () => {
    expect(getLineChartData(state)).toHaveLength(state.chart.data.length)
  })

  it('returns null if no data', () => {
    expect(getLineChartData(emptyState, 5)).toBe(null)
  })
})
