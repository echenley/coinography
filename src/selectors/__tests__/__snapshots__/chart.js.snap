// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCandleChartData returns candle chart data 1`] = `
Array [
  Object {
    "close": 11451.85,
    "high": 11469,
    "low": 11407.15,
    "open": 11407.16,
    "x": 2018-03-03T23:25:00.000Z,
  },
  Object {
    "close": 11423.99,
    "high": 11472.99,
    "low": 11417,
    "open": 11472.99,
    "x": 2018-03-03T23:30:00.000Z,
  },
  Object {
    "close": 11432,
    "high": 11437,
    "low": 11423.98,
    "open": 11423.99,
    "x": 2018-03-03T23:35:00.000Z,
  },
  Object {
    "close": 11418.99,
    "high": 11432.5,
    "low": 11402.09,
    "open": 11431.99,
    "x": 2018-03-03T23:40:00.000Z,
  },
  Object {
    "close": 11433,
    "high": 11433,
    "low": 11408.47,
    "open": 11419,
    "x": 2018-03-03T23:45:00.000Z,
  },
]
`;

exports[`getLineChartData returns line chart data 1`] = `
Array [
  Object {
    "x": 2018-03-03T23:25:00.000Z,
    "y": 11451.85,
  },
  Object {
    "x": 2018-03-03T23:30:00.000Z,
    "y": 11423.99,
  },
  Object {
    "x": 2018-03-03T23:35:00.000Z,
    "y": 11432,
  },
  Object {
    "x": 2018-03-03T23:40:00.000Z,
    "y": 11418.99,
  },
  Object {
    "x": 2018-03-03T23:45:00.000Z,
    "y": 11433,
  },
]
`;
